src/bindings/event.ts

```diff
@@ -15,14 +14,14 @@ export const getBotContext = () => botContext;
 let botConfig: BotConfig = {} as BotConfig;
 export const getBotConfig = () => botConfig;
 
-let adapters: Adapters = {} as Adapters;
+let adapters = {} as ReturnType<typeof createAdapters>;
 export const getAdapters = () => adapters;
 
 export type Logger = {
-  info: (msg: string | object, options?: JSON) => void;
-  debug: (msg: string | object, options?: JSON) => void;
-  warn: (msg: string | object, options?: JSON) => void;
-  error: (msg: string | object, options?: JSON) => void;
+  info: (msg: string, options?: JSON) => string;
+  debug: (msg: string, options?: JSON) => string;
+  warn: (msg: string, options?: JSON) => string;
+  error: (msg: string, options?: JSON) => string;
```

Comment:
Now we can `throw` and `return` logger.xxx() without being redundant and copying the string twice.